src/databricks/labs/dqx/engine.py

```diff
@@ -535,23 +612,50 @@ def apply_checks_by_metadata_and_save_in_table(
             input_config: Input configuration (e.g., table/view or file location and read options).
             output_config: Output configuration (e.g., table name, mode, and write options).
             quarantine_config: Optional configuration for writing invalid records.
+            metrics_config: Optional configuration for writing summary metrics.
             custom_check_functions: Optional mapping of custom check function names
                 to callables/modules (e.g., globals()).
             ref_dfs: Optional reference DataFrames used by checks.
         """
         # Read data from the specified table
         df = read_input_data(self.spark, input_config)
 
+        if metrics_config and isinstance(self._engine, DQEngineCore) and self._engine.observer and df.isStreaming:
```

Comment:
maybe encapsulate in common private method as we the same code in multiple places